src/libraries/Common/src/System/Net/IPv4AddressHelper.Common.cs

```diff
@@ -275,8 +304,11 @@ internal static unsafe long ParseNonCanonical(char* name, int start, ref int end
             {
                 // end of string, allowed
             }
-            else if ((ch = name[current]) == '/' || ch == '\\' || (notImplicitFile && (ch == ':' || ch == '?' || ch == '#')))
+            else if ((ch = ToUShort(name[current])) == '/' || ch == '\\' || (notImplicitFile && (ch == ':' || ch == '?' || ch == '#')))
```

Comment:
```suggestion
            else if (ch == '/' || ch == '\\' || (notImplicitFile && (ch == ':' || ch == '?' || ch == '#')))
```
Since you've made this type of optimization elsewhere, if I read it right, we shouldn't have to reread the value here either.

Reply:
I agree - thanks again. `current` is incremented without rereading `ch` when processing the hex prefix, but that situation's handled in the preceding check.